app/controllers/admin/members_controller.rb

```diff
@@ -1,12 +1,44 @@
+require "csv"
+
 module Admin
   class MembersController < ::MembersController
-    http_basic_authenticate_with MembershipRegister.basic_auth_options
-
-    respond_to :csv, only: :index
+    http_basic_authenticate_with(
+      name: ENV.fetch("AUTH_NAME") { "secretary" },
+      password: ENV.fetch("AUTH_PASSWORD") {
+        password = SecureRandom.hex
+        Rails.logger.info("You have not set a password for the admin area, your temporary password is: #{password}")
+        password
+      }
+    )
 
     def index
       @members = Member.all.order(updated_at: :desc)
-      respond_with @members
+
+      respond_to do |format|
+        format.html
+        format.csv do
+          csv_content = CSV.generate(headers: true) do |csv|
+            csv << %w[id full_name joined_at email address data created_at updated_at token token_updated_at]
+
+            @members.each do |member|
+              csv << [
+                member.id,
+                member.full_name,
+                member.joined_at,
+                member.email,
+                member.address,
+                member.data,
+                member.created_at,
+                member.updated_at,
+                member.token,
```

Comment:
What is the `token` used for? Do we want to expose these in the export?

Reply:
> What is the `token` used for? Do we want to expose these in the export?

I don't know I am only moving code along. I was not happy with it either.

Removing it :)

Reply:
Even the order of fields I kept the same from the previous implementation.